src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -925,6 +938,10 @@ void PasswordSafeFrame::RefreshToolbarButtons()
         toolbar->AddSeparator()->SetId(PwsToolbarButton.id);
     }
     else {
+      if ((PwsToolbarButton.id == ID_LOCK_SAFE) && m_sysTray->IsLocked())
+        continue; // When database is locked don't add the lock icon...
+      if ((PwsToolbarButton.id == ID_UNLOCK_SAFE) && !m_sysTray->IsLocked())
+        continue; // accordingly, don't add the unlock icon when database is unlocked
```

Comment:
Not sure I see the point of an Unlock button since the menubar is inaccessible behind the unlock modal. Alternatively, we could change the button's behavior to be a toggle and have it change the text according to current DB state.

Reply:
We have a Lock/Unlock menu item in the menu bar, and we have a Lock/Unlock button/tool ​​item in the toolbar, which is located below the menu bar.
When the database is locked, the application is minimized. In a Wayland environment, the application can be restored or brought to the foreground without the unlock dialog appearing. In this case the user should be able to unlock the database via the menu as well as the toolbar.
Depending on the database status (locked/unlocked), the corresponding menu item and toolbar item should have the appropriate label.